buildspec.yml

```diff
@@ -0,0 +1,33 @@
+version: 0.2
+
+phases:
+  install:
+    commands:
+      - sudo apt-get update
+      - sudo apt-get install apt-transport-https
+      - curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+      - echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+      - sudo apt-get update && sudo apt-get install yarn
+  pre_build:
+    commands:
+      - yarn install
+  build:
+    commands:
+      - yarn build:all
+      - ./node_modules/mocha/bin/mocha -r ts-node/register test/**/*.spec.ts
+  post_build:
+    commands:
+      - >-
+        if [ $CODEBUILD_SOURCE_VERSION == master ]; then
```

Comment:
We might have to put this stuff in a shell script.  I'm seeing this in the logs:
<img width="1220" alt="screen shot 2018-02-21 at 1 08 58 pm" src="https://user-images.githubusercontent.com/3791752/36508770-797a66d8-1712-11e8-9211-fbcbb683aecf.png">

Reply:
Huh, I'll have to keep an eye on those logs, I didn't notice that. Well, it didn't result in an error for me. I wonder if the logical AND may be causing the issue? The shell may be thinking, "then -> do this STOP && do this thing", after the conditional. But, I figured the semi-colon and `fi` would demarcate that, so I didn't bother grouping with parentheses there.

Reply:
Could be.  I'd bet it has something to do with the parser and putting that stuff all in a yml file.  I'd bet if we put it in a shell script and just ran that script post_build it would take care of deployments and not choke on any syntax.